src/lib/message/L2ToL1Message.ts

```diff
@@ -312,6 +315,71 @@ export class L2ToL1MessageReader extends L2ToL1Message {
       await this.waitUntilOutboxEntryCreated(retryDelay)
     }
   }
+
+  // TODO: create version that queries multiple L2 to L1 txs, so a single multicall can make all requests
+  public async getExpectedNodeDeadline(
```

Comment:
Maybe we could have a more general name for this? Maybe `getFirstExecutableBlock` or something like that? And include in the comment that this is the deadline of the node which contains this message.

Reply:
yup good point, then we should probably add a bit of padding to the return value to account for the time difference between a node being confirmable, and a validator actually confirming it